Mathlib/AlgebraicTopology/FundamentalGroupoid/SimplyConnected.lean

```diff
@@ -42,7 +42,9 @@ class SimplyConnectedSpace (X : Type*) [TopologicalSpace X] : Prop where
 theorem simply_connected_iff_unique_homotopic (X : Type*) [TopologicalSpace X] :
     SimplyConnectedSpace X ↔
       Nonempty X ∧ ∀ x y : X, Nonempty (Unique (Path.Homotopic.Quotient x y)) := by
-  rw [simply_connected_def, equiv_punit_iff_unique]; rfl
+  rw [simply_connected_def, equiv_punit_iff_unique]
+  refine ⟨fun ⟨⟨x⟩,h⟩ => ⟨⟨x.as⟩, fun x y => h _ _⟩,
+    fun ⟨⟨x⟩,h⟩ => ⟨⟨⟨x⟩⟩, fun x y => h _ _⟩ ⟩
```

Comment:
Is there a missing lemma here? Maybe `Nonempty (FundamentalGroupoid X) \iff Nonempty X` (and similarly for `Subsingleton` and `Nontrivial`)

Reply:
(the trick is to build the equiv and then all these results are there by dot notation)

Reply:
Probably. I didn't think too much when I made these changes :) I'll fix this soon.

Reply:
I added `equiv` and some related lemmas and instances. I don't want this PR to expand too far beyond the original fix, so I would prefer to leave anything further which is not directly related to a future PR.